src/spaces/openstreetmap.jl

```diff
@@ -0,0 +1,591 @@
+export OpenStreetMapSpace, OSMSpace, TEST_MAP
+export osm_random_road_position,
+    osm_plan_route,
+    osm_map_coordinates,
+    osm_road_length,
+    osm_is_stationary,
+    osm_random_route!
+
+struct OpenStreetMapSpace <: DiscreteSpace
+    m::OpenStreetMapX.MapData
+    edges::Vector{CartesianIndex{2}}
+    s::Vector{Vector{Int}}
+end
+
+"""
+    OpenStreetMapSpace(path::AbstractString; kwargs...)
+Create a space residing on the Open Street Map (OSM) file provided via `path`.
+
+The abbreviation `OSMSpace` may be used interchangeably.
+
+Much of the functionality of this space is provided by interfacing with
+[OpenStreetMapX.jl](https://github.com/pszufe/OpenStreetMapX.jl), for example the two
+keyword arguments `use_cache = false` and `trim_to_connected_graph = true` can be
+passed into the [`OpenStreetMapX.get_map_data`](@ref) function.
+
+For details on how to obtain an OSM file for your use case, consult the OpenStreetMapX.jl
+README. We provide a variable `TEST_MAP` to use as a `path` for testing.
+
+This space represents the underlying map as a *continuous* entity choosing accuracy over
+performance. If your solution can tolerate routes to and from intersections only, a
+faster implementation can be achieved by using the
+[graph representation](https://pszufe.github.io/OpenStreetMapX.jl/stable/reference/#OpenStreetMapX.MapData)
+of your map provided by OpenStreetMapX.jl. For tips on how to implement this, see our
+integration example: [Social networks with LightGraphs.jl](@ref).
+
```

Comment:
We link to an example for using the light graphs, but we should link to an example that _actually_ uses the space as well, the zombie apocalypse. In fact, I think that in all spaces, after the opening sentence of the docstring we link to an example directly.